Thanks  @vincbeck ! I have tested this modification on my Helm Chart installed image based on `apache/airflow:3.0.2-python3.9`

By applying this modification, I firstly got a 500 Internal Server Error:

```
INFO:     {masked ip}- "POST /auth/token HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/airflow/.local/lib/python3.9/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/gzip.py", line 29, in __call__
    await responder(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/gzip.py", line 126, in __call__
    await super().__call__(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/gzip.py", line 46, in __call__
    await self.app(scope, receive, self.send_with_compression)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/cors.py", line 85, in __call__
    await self.app(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/base.py", line 183, in __call__
    raise app_exc
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/base.py", line 141, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 460, in handle
    await self.app(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
  File "/home/airflow/.local/lib/python3.9/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
  File "/home/airflow/.local/lib/python3.9/site-packages/fastapi/routing.py", line 214, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/home/airflow/.local/lib/python3.9/site-packages/starlette/concurrency.py", line 37, in run_in_threadpool
    return await anyio.to_thread.run_sync(func)
  File "/home/airflow/.local/lib/python3.9/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/home/airflow/.local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 2470, in run_sync_in_worker_thread
    return await future
  File "/home/airflow/.local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 967, in run
    result = context.run(func, *args)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/fab/auth_manager/api_fastapi/routes/login.py", line 38, in create_token
    return FABAuthManagerLogin.create_token(body=body)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/fab/auth_manager/api_fastapi/services/login.py", line 51, in create_token
    user = auth_manager.security_manager.auth_user_ldap(body.username, body.password)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/fab/auth_manager/security_manager/override.py", line 1885, in auth_user_ldap
    self._rotate_session_id()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/fab/auth_manager/security_manager/override.py", line 2191, in _rotate_session_id
    session.sid = str(uuid.uuid4())
  File "/home/airflow/.local/lib/python3.9/site-packages/werkzeug/local.py", line 316, in __get__
    obj = instance._get_current_object()
  File "/home/airflow/.local/lib/python3.9/site-packages/werkzeug/local.py", line 513, in _get_current_object
    raise RuntimeError(unbound_message) from None
RuntimeError: Working outside of request context.
This typically means that you attempted to use functionality that needed
an active HTTP request. Consult the documentation on testing for
information about how to avoid this problem.
```

and then I modified the login.py and override.py by adding the rotate_session_id=False also to `auth_user_ldap` and now it works! (Actually I'm not quite sure what is it for and why you set that for the auth_user_db while the original version will always do this stuff)

```
# login.py

        if auth_manager.security_manager.auth_type == AUTH_LDAP:
            user = auth_manager.security_manager.auth_user_ldap(body.username, body.password, rotate_session_id=False)
        if user is None:
            user = auth_manager.security_manager.auth_user_db(body.username, body.password, rotate_session_id=False)
```

```
# override.py

            # LOGIN SUCCESS (only if user is now registered)
            if user:
                if rotate_session_id:
                    self._rotate_session_id()
                self.update_user_auth_stat(user)
                return user
```

---

Also FYI, here is my test image. 

```
# 3.0.2
FROM apache/airflow:3.0.2-python3.9

# Got version conflict on kerberos so not using all-core
# RUN pip install 'apache-airflow[all-core]==3.0.2'
# --- core ---
RUN pip install 'apache-airflow[sentry]==3.0.2'
RUN pip install 'apache-airflow[otel]==3.0.2'
RUN pip install 'apache-airflow[standard]==3.0.2'
RUN pip install 'apache-airflow[graphviz]==3.0.2'
RUN pip install 'apache-airflow[async]==3.0.2'
RUN pip install 'apache-airflow[s3fs]==3.0.0'

RUN pip install 'apache-airflow[edge3]==3.0.2'
RUN pip install 'apache-airflow[ldap]==3.0.2'
RUN pip install 'apache-airflow[postgres]==3.0.2'
RUN pip install 'apache-airflow[mysql]==3.0.2'
RUN pip install 'apache-airflow[celery]==3.0.2'
RUN pip install 'apache-airflow[apache-kafka]==3.0.2'
RUN pip install 'apache-airflow[ssh]==3.0.2'

COPY login.py /home/airflow/.local/lib/python3.9/site-packages/airflow/providers/fab/auth_manager/api_fastapi/services/login.py
COPY override.py /home/airflow/.local/lib/python3.9/site-packages/airflow/providers/fab/auth_manager/security_manager/override.py
```

and this is the sever config in my helm chat values.yaml

(If it is recommended, I can start another issue to decribe this) I was using a third-party airflow helm chart on Airflow 2.* but currently using the official Helm Chart 1.17 this time. I notice that the webServer section in official chart seems not being used at all, only apiServer got used (not sure if it is the expected behavior). For example, no `webServer` pod is deployed even it is enabled, and webServerConfig is not used. I looked into the values.yml and decided to move all web-server stuffs to api-server then everything looks fine (yeah. including the LDAP settings).

```
apiServer:
  apiServerConfig: |-
    from flask_appbuilder.security.manager import AUTH_LDAP

    from airflow import configuration as conf

    CSRF_ENABLED = True

    AUTH_TYPE = AUTH_LDAP
    AUTH_ROLE_ADMIN = "Admin"
    AUTH_LDAP_SERVER = "<MASKED>"
    AUTH_LDAP_USE_TLS = False
    AUTH_LDAP_START_TLS = False
    AUTH_LDAP_SSL_SKIP_VERIFY = True

    # registration configs
    AUTH_USER_REGISTRATION = True
    AUTH_USER_REGISTRATION_ROLE = "User"
    AUTH_LDAP_FIRSTNAME_FIELD = "givenName"
    AUTH_LDAP_LASTNAME_FIELD = "sn"
    AUTH_LDAP_EMAIL_FIELD = "mail"

    # search configs
    AUTH_LDAP_SEARCH = "dc=<MASKED>,dc=com"
    AUTH_LDAP_SEARCH_BASE_DNS = "dc=<MASKED>,dc=com"
    AUTH_LDAP_UID_FIELD = 'sAMAccountName'
    AUTH_LDAP_BIND_USER = <MASKED>
    AUTH_LDAP_BIND_PASSWORD = <MASKED>

    # mapping from LDAP DN to airflow roles
    AUTH_ROLES_MAPPING = {
        'CN=<MASKED>,OU=Groups_Local,DC=<MASKED>,DC=com': ["Admin"],
    }
    AUTH_LDAP_GROUP_FIELD = "memberOf"
    AUTH_ROLES_SYNC_AT_LOGIN = True
    PERMANENT_SESSION_LIFETIME = 1800

    # LDAPS
    AUTH_LDAP_ALLOW_SELF_SIGNED = True
    AUTH_LDAP_TLS_CACERTFILE = '/etc/ssl/certs/ca-certificates.crt'
```

Contact me if you need any futher information!

--- a/providers/fab/docs/auth-manager/token.rst
+++ b/providers/fab/docs/auth-manager/token.rst
@@ -40,3 +40,6 @@ Example
         }'
 
 This process will return a token that you can use in the Airflow public API requests.
+
+Only users from database (`AUTH_TYPE = AUTH_DB`) or from LDAP (`AUTH_TYPE = AUTH_LDAP`) can be used to generate a token.
+See :doc:`Airflow public API <webserver-authentication>` for more details.

--- a/providers/fab/src/airflow/providers/fab/auth_manager/api_fastapi/services/login.py
+++ b/providers/fab/src/airflow/providers/fab/auth_manager/api_fastapi/services/login.py
@@ -18,6 +18,7 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING, cast
 
+from flask_appbuilder.const import AUTH_LDAP
 from starlette import status
 from starlette.exceptions import HTTPException
 
@@ -44,14 +45,22 @@ class FABAuthManagerLogin:
             )
 
         auth_manager = cast("FabAuthManager", get_auth_manager())
-        user: User = auth_manager.security_manager.find_user(username=body.username)
-        if not user:
-            raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid username")
+        user: User | None = None
 
-        if auth_manager.security_manager.check_password(username=body.username, password=body.password):
-            return LoginResponse(
-                access_token=auth_manager.generate_jwt(
-                    user=user, expiration_time_in_seconds=expiration_time_in_seconds
-                )
+        if auth_manager.security_manager.auth_type == AUTH_LDAP:
+            user = auth_manager.security_manager.auth_user_ldap(
+                body.username, body.password, rotate_session_id=False
             )
-        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid password")
+        if user is None:
+            user = auth_manager.security_manager.auth_user_db(
+                body.username, body.password, rotate_session_id=False
+            )
+
+        if not user:
+            raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid credentials")
+
+        return LoginResponse(
+            access_token=auth_manager.generate_jwt(
+                user=user, expiration_time_in_seconds=expiration_time_in_seconds
+            )
+        )

--- a/providers/fab/src/airflow/providers/fab/auth_manager/security_manager/override.py
+++ b/providers/fab/src/airflow/providers/fab/auth_manager/security_manager/override.py
@@ -1723,7 +1723,7 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
     --------------------
     """
 
-    def auth_user_ldap(self, username, password):
+    def auth_user_ldap(self, username, password, rotate_session_id=True):
         """
         Authenticate user with LDAP.
 
@@ -1890,7 +1890,8 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
 
             # LOGIN SUCCESS (only if user is now registered)
             if user:
-                self._rotate_session_id()
+                if rotate_session_id:
+                    self._rotate_session_id()
                 self.update_user_auth_stat(user)
                 return user
             return None
@@ -1919,7 +1920,7 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
             return False
         return check_password_hash(user.password, password)
 
-    def auth_user_db(self, username, password):
+    def auth_user_db(self, username, password, rotate_session_id=True):
         """
         Authenticate user, auth db style.
 
@@ -1927,6 +1928,8 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
             The username or registered email address
         :param password:
             The password, will be tested against hashed password on db
+        :param rotate_session_id:
+            Whether to rotate the session ID
         """
         if username is None or username == "":
             return None
@@ -1942,7 +1945,8 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
             log.info(LOGMSG_WAR_SEC_LOGIN_FAILED, username)
             return None
         if check_password_hash(user.password, password):
-            self._rotate_session_id()
+            if rotate_session_id:
+                self._rotate_session_id()
             self.update_user_auth_stat(user, True)
             return user
         self.update_user_auth_stat(user, False)

--- a/providers/fab/tests/unit/fab/auth_manager/api_fastapi/services/test_login.py
+++ b/providers/fab/tests/unit/fab/auth_manager/api_fastapi/services/test_login.py
@@ -17,16 +17,30 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
-from unittest.mock import MagicMock, patch
+from unittest.mock import ANY, MagicMock, patch
 
 import pytest
+from flask_appbuilder.const import AUTH_DB, AUTH_LDAP, AUTH_OID
 from starlette.exceptions import HTTPException
 
 from airflow.providers.fab.auth_manager.api_fastapi.services.login import FABAuthManagerLogin
 
-if TYPE_CHECKING:
-    from airflow.providers.fab.auth_manager.api_fastapi.datamodels.login import LoginResponse
+
+@pytest.fixture
+def auth_manager():
+    return MagicMock()
+
+
+@pytest.fixture
+def security_manager():
+    return MagicMock()
+
+
+@pytest.fixture
+def user():
+    user = MagicMock()
+    user.password = "dummy"
+    return user
 
 
 @patch("airflow.providers.fab.auth_manager.api_fastapi.services.login.get_auth_manager")
@@ -34,71 +48,56 @@ class TestLogin:
     def setup_method(
         self,
     ):
-        self.dummy_auth_manager = MagicMock()
-        self.dummy_app_builder = MagicMock()
-        self.dummy_app = MagicMock()
-        self.dummy_login_body = MagicMock()
-        self.dummy_user = MagicMock()
-        self.dummy_user.password = "dummy"
-        self.dummy_security_manager = MagicMock()
-        self.dummy_login_body.username = "dummy"
-        self.dummy_login_body.password = "dummy"
+        self.login_body = MagicMock()
+        self.login_body.username = "username"
+        self.login_body.password = "password"
         self.dummy_token = "DUMMY_TOKEN"
 
-    def test_create_token(self, get_auth_manager):
-        get_auth_manager.return_value = self.dummy_auth_manager
-        self.dummy_auth_manager.security_manager = self.dummy_security_manager
-        self.dummy_security_manager.find_user.return_value = self.dummy_user
-        self.dummy_auth_manager.generate_jwt.return_value = self.dummy_token
-        self.dummy_security_manager.check_password.return_value = True
+    @pytest.mark.parametrize(
+        "auth_type, method",
+        [
+            [AUTH_DB, "auth_user_db"],
+            [AUTH_LDAP, "auth_user_ldap"],
+        ],
+    )
+    def test_create_token(self, get_auth_manager, auth_type, method, auth_manager, security_manager, user):
+        security_manager.auth_type = auth_type
+        getattr(security_manager, method).return_value = user
 
-        login_response: LoginResponse = FABAuthManagerLogin.create_token(
-            body=self.dummy_login_body,
-            expiration_time_in_seconds=1,
+        auth_manager.security_manager = security_manager
+        auth_manager.generate_jwt.return_value = self.dummy_token
+
+        get_auth_manager.return_value = auth_manager
+
+        result = FABAuthManagerLogin.create_token(
+            body=self.login_body,
         )
-        assert login_response.access_token == self.dummy_token
+        assert result.access_token == self.dummy_token
+        getattr(security_manager, method).assert_called_once_with(
+            self.login_body.username, self.login_body.password, rotate_session_id=False
+        )
+        auth_manager.generate_jwt.assert_called_once_with(user=user, expiration_time_in_seconds=ANY)
 
-    def test_create_token_invalid_username(self, get_auth_manager):
-        get_auth_manager.return_value = self.dummy_auth_manager
-        self.dummy_auth_manager.security_manager = self.dummy_security_manager
-        self.dummy_security_manager.find_user.return_value = None
-        self.dummy_security_manager.check_password.return_value = False
+    @pytest.mark.parametrize(
+        "auth_type, methods",
+        [
+            [AUTH_DB, ["auth_user_db"]],
+            [AUTH_LDAP, ["auth_user_ldap", "auth_user_db"]],
+            [AUTH_OID, ["auth_user_db"]],
+        ],
+    )
+    def test_create_token_no_user(
+        self, get_auth_manager, auth_type, methods, auth_manager, security_manager, user
+    ):
+        security_manager.auth_type = auth_type
+        for method in methods:
+            getattr(security_manager, method).return_value = None
+
+        auth_manager.security_manager = security_manager
+        get_auth_manager.return_value = auth_manager
 
         with pytest.raises(HTTPException) as ex:
             FABAuthManagerLogin.create_token(
-                body=self.dummy_login_body,
-                expiration_time_in_seconds=1,
+                body=self.login_body,
             )
         assert ex.value.status_code == 401
-        assert ex.value.detail == "Invalid username"
-
-    def test_create_token_invalid_password(self, get_auth_manager):
-        get_auth_manager.return_value = self.dummy_auth_manager
-        self.dummy_auth_manager.security_manager = self.dummy_security_manager
-        self.dummy_security_manager.find_user.return_value = self.dummy_user
-        self.dummy_user.password = "invalid_password"
-        self.dummy_security_manager.check_password.return_value = False
-
-        with pytest.raises(HTTPException) as ex:
-            FABAuthManagerLogin.create_token(
-                body=self.dummy_login_body,
-                expiration_time_in_seconds=1,
-            )
-        assert ex.value.status_code == 401
-        assert ex.value.detail == "Invalid password"
-
-    def test_create_token_empty_user_password(self, get_auth_manager):
-        get_auth_manager.return_value = self.dummy_auth_manager
-        self.dummy_auth_manager.security_manager = self.dummy_security_manager
-        self.dummy_security_manager.find_user.return_value = self.dummy_user
-        self.dummy_login_body.username = ""
-        self.dummy_login_body.password = ""
-        self.dummy_security_manager.check_password.return_value = False
-
-        with pytest.raises(HTTPException) as ex:
-            FABAuthManagerLogin.create_token(
-                body=self.dummy_login_body,
-                expiration_time_in_seconds=1,
-            )
-        assert ex.value.status_code == 400
-        assert ex.value.detail == "Username and password must be provided"

